common/meta/types/proto/request.proto

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 Datafuse Labs.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package meta;
+
+message SeqV {
+  uint64 seq = 1;
+  bytes data = 2;
+}
+
+message TxnKey {
+  string key = 1;
+  string tenant = 2;
+  string tree = 3;
+}
+
+// Get request and response
+message GetRequest {
+  TxnKey key = 1;
+}
+
+message GetResponse {
+  TxnKey key = 1;
+
+  optional SeqV value = 2;
+}
+
+// Put request and response
+message PutRequest {
+  TxnKey key = 1;
+  bytes value = 2;
+  bool prev_kv = 3;
```

Comment:
What does this `bool` mean?

Reply:
> What does this `bool` mean?

if or not return prev value

Reply:
Maybe always returning the previous kv would be better?
The application does not need to think about whether it needs it or not.
And the response handler in the application does not need to know whether the method that sends the request has set this field.

Reply:
i just follow etcd api define, how about let transaction API user to judge these param

Reply:
OK. your code your decision.